src/NuGet.Core/NuGet.Packaging/Signing/Content/KeyPairFileReader.cs

```diff
@@ -33,86 +31,70 @@ public KeyPairFileReader(Stream stream)
         /// <remarks>Returns an empty set if the file has reached the end.</remarks>
         public Dictionary<string, string> ReadSection()
         {
-            var hasEntries = false;
             var entries = new Dictionary<string, string>(StringComparer.Ordinal);
 
             var line = _reader.ReadLine();
-            while (line != null)
-            {
-                if (line.Length == 0)
-                {
-                    // End of section
-                    if (!hasEntries)
-                    {
-                        // Empty sections are invalid
-                        ThrowInvalidFormat();
-                    }
 
-                    break;
-                }
+            if (line == null)
+            {
+                return entries;
+            }
 
-                hasEntries = true;
-                var pair = GetPair(line);
-                var key = pair.Key;
+            while (!string.IsNullOrEmpty(line))
+            {
+                var property = GetProperty(line);
 
-                if (!entries.ContainsKey(key))
+                if (entries.ContainsKey(property.Key))
                 {
-                    entries.Add(key, pair.Value);
+                    ThrowInvalidFormat();
                 }
                 else
                 {
-                    // Key is not allowed or a duplicate
-                    ThrowInvalidFormat();
+                    entries.Add(property.Key, property.Value);
                 }
 
                 line = _reader.ReadLine();
             }
 
+            // Read section break.
+            if (line != string.Empty)
```

Comment:
Will this code ever be executed? I might be missing something, but it seems like if the code has finished executing the `while` loop above, then `line` will always be empty.

Reply:
It will be executed.  The `while` loop will exit if `line` is `null` or an empty string.  If `line` is `null`, it means there was no line; we reached the end of the file.  If `line` is an empty string, it means that it was an empty line (e.g.:  \n or \r\n).  Sections are terminated with a new line.  There are [tests](https://github.com/NuGet/NuGet.Client/pull/1838/files#diff-13e795da821fb256871fe7e5dabb4197R55) for both of these scenarios.